### PR TITLE
COCOS-151 - Add compression/decompression option for CLI/Agent

### LINKED ISSUE #151: Feature: Add compression/decompression option for CLI/Agent
### Is your feature request related to a problem? Please describe.

The data that is being sent to the Agent sometimes needs to be compressed. This should be introduced as a new feature into CLI and the Agent should be modified to be able to decompress the data.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Add an optional flag to `data` option of the CLI to compress and send data to the Agent. The Agent should then decompress the data and use it with the algorithm.

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SammyOina**
# What type of PR is this?
This is a feature that allows on the fly compression and decompression

## What does this do?
- **New Features**
	- Enhanced command-line interface with simplified commands and new options for algorithm and dataset uploads, including checksum validation.
	- Support for in-memory dataset zipping and enhanced context management for decompression flags.

- **Bug Fixes**
	- Improved error handling for checksum calculations and command executions.

- **Documentation**
	- Updated CLI documentation to reflect command changes and new features.

- **Refactor**
	- Streamlined checksum calculations and context management to improve performance and clarity.

## Which issue(s) does this PR fix/relate to?

- Resolves #151 

## Have you included tests for your changes?
No

## Did you document any new/modified feature?
Yes

### Notes

```shell
go run ./cmd/cli/main.go data ./test/manual/data private.pem -d  
```

```shell
# ls /datasets
iris.csv
```

### CODE REVIEW

**Review by WashingtonKK — CHANGES_REQUESTED** [review]
```
➜  cocos git:(cocos-151) ✗ ./build/cocos-cli data ./test/manual/data private.pem -d
2024/08/16 12:29:55 Uploading dataset CSV: ./test/manual/data
Uploading data...  100% [====================================================>] 
{"time":"2024-08-16T12:29:55.989831624+03:00","level":"ERROR","msg":"Failed to send Data"}
2024/08/16 12:29:55 Error uploading dataset: rpc error: code = Unknown desc = malformed data, filename does not match manifest

```

image

How do you handle and verify file name in this structure since it does not work

**Comment by SammyOina** [comment]
> ```
> 
> ➜  cocos git:(cocos-151) ✗ ./build/cocos-cli data ./test/manual/data private.pem -d
> 
> 2024/08/16 12:29:55 Uploading dataset CSV: ./test/manual/data
> 
> Uploading data...  100% [====================================================>] 
> 
> {"time":"2024-08-16T12:29:55.989831624+03:00","level":"ERROR","msg":"Failed to send Data"}
> 
> 2024/08/16 12:29:55 Error uploading dataset: rpc error: code = Unknown desc = malformed data, filename does not match manifest
> 
> 
> 
> ```
> 
> 
> 
> image
> 
> 
> 
> How do you handle and verify file name in this structure since it does not work

Use the base name of the directory, check code to see how directories are handled

**Review by WashingtonKK — CHANGES_REQUESTED** [review]
This scheme works with the test computations server but not with prism, please check where the issue lies.

File: cli/datasets.go [hunk 1]
```diff
@@ -11,27 +12,45 @@ import (
 
 	"github.com/spf13/cobra"
 	"github.com/ultravioletrs/cocos/agent"
+	"github.com/ultravioletrs/cocos/internal"
+	"google.golang.org/grpc/metadata"
 )
 
+var decompressDataset bool
```
[the remark this answers is not in the document]
Comment (SammyOina) [reply cli/datasets.go:19]: No

**Comment by SammyOina** [comment]
> This scheme works with the test computations server but not with prism, please check where the issue lies.

Which scheme, and what does not work?

File: cli/datasets.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (SammyOina) [reply cli/datasets.go:19]: this states whether to decompress data on arrival or leave it compressed

**Comment by SammyOina** [comment]
> ```
> ➜  cocos git:(cocos-151) ✗ ./build/cocos-cli data ./test/manual/data private.pem -d
> 2024/08/16 12:29:55 Uploading dataset CSV: ./test/manual/data
> Uploading data...  100% [====================================================>] 
> {"time":"2024-08-16T12:29:55.989831624+03:00","level":"ERROR","msg":"Failed to send Data"}
> 2024/08/16 12:29:55 Error uploading dataset: rpc error: code = Unknown desc = malformed data, filename does not match manifest
> ```
> 
> image
> 
> How do you handle and verify file name in this structure since it does not work

base name here will be `data` if you can refer here https://github.com/SammyOina/cocos-ai/blob/1cde120d705e73cdc7fdb7993011804ba1da8c65/cli/datasets.go#L53

**Review by WashingtonKK — APPROVED** [review]
FileName on prism needs to be: `data`

Please also update docs.

**Pull request merged**